src/tests/ftest/pool/pool_svc.py

```diff
@@ -4,85 +4,157 @@
 
-            try:
+            if svc_params[1] > 2:
                 self.pool.get_info()
                 leader = self.pool.info.pi_leader
-                if createsvc[0] == 3:
-                    # kill pool leader and exclude it
+                all_svc_ranks = list(self.pool.svc_ranks)
+                all_svc_ranks.remove(leader)
+                non_leader = all_svc_ranks[-1]
+
+                # Stop the pool leader
+                self.log.info("Stopping the pool leader: %s", leader)
+                try:
                     self.pool.pool.pool_svc_stop()
+                except DaosApiError as error:
+                    self.log.info(error)
+                    self.fail("Error issuing DaosPool.pool_svc_stop()")
+
+                # Exclude the pool leader
+                self.log.info("Excluding the pool leader: %s", leader)
+                try:
                     self.pool.exclude([leader], self.d_log)
-                    # perform pool disconnect, try connect again and disconnect
+                except TestFail as error:
+                    self.log.info(error)
+                    self.fail(
+                        "Error issuing TestPool.exclude([{}])".format(leader))
+                self.server_managers[-1].update_expected_states(
+                    [leader], "evicted")
+
+                # Disconnect from the pool
+                self.log.info("Disconnecting from the pool")
+                try:
                     self.pool.disconnect()
+                except TestFail as error:
+                    self.log.info(error)
+                    self.fail(
+                        "Error issuing TestPool.disconnect() after excluding "
+                        "the leader")
+
+                # Connect to the pool
+                self.log.info("Connecting from the pool")
+                try:
                     self.pool.connect()
+                except TestFail as error:
+                    self.log.info(error)
+                    self.fail("Error issuing TestPool.connect()")
+
+                # Disconnect from the pool
+                self.log.info("Disconnecting from the pool")
+                try:
                     self.pool.disconnect()
-                    # kill another server which is not a leader and exclude it
-                    self.server_managers[0].stop_ranks([3], self.test_log)
-                    self.pool.exclude([3], self.d_log)
-                    # perform pool connect
+                except TestFail as error:
+                    self.log.info(error)
+                    self.fail(
+                        "Error issuing TestPool.disconnect() after connecting "
+                        "to the pool")
+
+                # Stop a pool non-leader
+                self.log.info("Stopping a pool non-leader: %s", non_leader)
+                try:
+                    self.pool.start_rebuild([non_leader], self.test_log)
```

Comment:
(pylint-no-member) Instance of 'TestPool' has no 'start_rebuild' member

Reply:
Fixed